Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityToolkitFiles.cs

```diff
@@ -408,14 +408,9 @@ private static void RegisterFolderToModule(string folderPath, MixedRealityToolki
             }
 
             modFolders.Add(normalizedFolder);
-
-            if (module == MixedRealityToolkitModuleType.Core)
-            {
-                TryToCreateGeneratedFolder(folderPath);
-            }
         }
 
-        private static void TryToCreateGeneratedFolder(string folderPath)
+        public static void TryToCreateGeneratedFolder(string folderPath)
```

Comment:
Is there potentially a way to reuse the existing path (I think it's https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/e888c7a91ee1d8383ccd4c9a6b8d40b9ce85c4bf/Assets/MixedRealityToolkit/Utilities/Editor/Preferences/ProjectPreferences.cs#L56) here?

Reply:
Though also the lines you removed above seem unexpected (create the generated folder not when the module is `Generated`, but `Core`). @Troy-Ferrell?

Reply:
Maybe instead of having this method take in a folder path, it just defaults to `Assets` in all cases?

Reply:
So there seems like quite a lot of conflicts here. Things we need to resolve
1) Does MRTK.Generated always exist at directly underneath assets...or should it be placed next to the rest of MRTK, originally designed as next to MRTK.Core?

2) The MRTKFiles.cs code needs to be fixed. I poorly wrote/fixed it. The if module == Core code, should be cleaned up  that on SearchFolders() we TryToCreateGeneratedFolder() and instead of a folder path parameter, we just use the right path in the function

My vote is we place MRTK.Generated wherever the user has placed the plugin(i.e next to MRTK.Core). if it doesn't exist. 

Of course, the MRTK.Generated folder could be moved by the user, in which case that is fine. We will track the sentinel file.

Reply:
Thanks @Troy-Ferrell 
@keveleigh @wiwei  Should we place the MRTK.Generated folder next to MRTK.Core or always at the root of Assets?

Reply:
>Does MRTK.Generated always exist at directly underneath assets

I think it should always exist under assets, for reasons described in #7184 (mostly around package distribution)

Reply:
I think under Assets as well.

Going where the Core is also has some trouble on other UPM stories (i.e. it would go into a non-mutable location)

Reply:
Ok, I'll update the change to Kurtis's suggestion to remove `folderPath` and always default to `Assets`

Reply:
@keveleigh ahh good point.

To be clear for all though, if no MRTK.Generated folder can be found, then one will be created under assets. 

The user could still move it and thus all MRTK code should check the MRTKFiles class for it's path. Thus the profile clone window still needs to remove its hardcoded paths